types/jsstore/index.d.ts

```diff
@@ -0,0 +1,705 @@
+    }
+}
+declare module JsStore {
+    module Business {
+        module Select {
+            class Where extends Like {
+                private executeWhereLogic;
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Select {
+            class Join extends BaseSelect {
+                Query: ISelectJoin;
+                QueryStack: Array<ITableJoin>;
+                CurrentQueryStackIndex: number;
+                private onTransactionCompleted;
+                private executeWhereJoinLogic;
+                private executeRightJoin;
+                private executeWhereUndefinedLogicForJoin;
+                private startExecutionJoinLogic();
+                constructor(query: ISelectJoin, onSuccess: Function, onError: Function);
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Select {
+            class Instance extends Where {
+                onTransactionCompleted: () => void;
+                private createtransactionForOrLogic;
+                private orQuerySuccess;
+                private executeOrLogic;
+                constructor(query: ISelect, onSuccess: Function, onError: Function);
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Count {
+            class BaseCount extends Base {
+                ResultCount: number;
+                SkipRecord: any;
+                LimitRecord: any;
+                CheckFlag: boolean;
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Count {
+            class NotWhere extends BaseCount {
+                protected executeWhereUndefinedLogic: () => void;
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Count {
+            class Like extends NotWhere {
+                CompSymbol: Occurence;
+                CompValue: any;
+                Column: any;
+                CompValueLength: Number;
+                private filterOnOccurence;
+                protected executeLikeLogic: (column: any, value: any, symbol: Occurence) => void;
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Count {
+            class Where extends Like {
+                private executeWhereLogic;
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Count {
+            class Instance extends Where {
+                onTransactionCompleted: () => void;
+                constructor(query: ICount, onSuccess: Function, onError: Function);
+            }
+        }
+    }
+}
+declare module JsStore {
+    module Business {
+        module Update {
+            var updateValue: (suppliedValue: any, storedValue: any) => any;
+        }
+        class BaseUpdate extends Base {
+            CheckFlag: boolean;
+        }
+    }
+}
+declare module JsStore {
```

Comment:
All of these re-openings of the module should be changed to a single declaration. This is adding a lot of noise for no reason - the file should be ~40% smaller than it is rightn ow